lib/strings.nix

```diff
@@ -126,6 +126,20 @@ rec {
     # List of input strings
     list: concatStringsSep sep (lib.imap1 f list);
 
+  /* Generate a string by repeating a given string a given number of times.
+
+     Type: repeat :: string -> int -> string
+
+     Example:
+       repeat "A" 3
+       => "AAA"
+  */
+  repeat =
```

Comment:
I think this might be more useful with the arguments round the other way?  I could imagine myself currying `repeat 3` more than `repeat "str"`.